### PR TITLE
Cloud client library migration part 2: Device creation / deletion

### PR DESCRIPTION
**Pull request by gguuss**
Migrates the create / delete device snippets to Cloud Client library

### CODE REVIEW

**Comment by gguuss** [comment]
/FYI It looks like the failing tests are outside of my change.

File: iot/manager/manager.js [hunk 1]
```diff
@@ -224,28 +222,27 @@ const createUnauthDevice = async (
...
+    // optional auth parameters.
+  });
 
+  const regPath = iotClient.registryPath(projectId, cloudRegion, registryId);
+  const device = {id: deviceId};
   const request = {
-    parent: registryName,
-    resource: {id: deviceId},
+    parent: regPath,
+    device: device,
```
Comment (fhinkel) [inline iot/manager/manager.js]: You can use shorthand notation here: 

```
const request = {
  parent: regPath,
  device
}

File: iot/manager/manager.js [hunk 2]
```diff
@@ -331,21 +329,17 @@ const createEsDevice = async (
...
-    );
-
-    console.log('Created device');
-    console.log(data);
+    const responses = await iotClient.createDevice(request);
+    const response = responses[0];
+    console.log('Created device', response);
   } catch (err) {
-    console.error('Could not create device', err);
+    console.error('Could note create device', err);
```
Comment (fhinkel) [inline iot/manager/manager.js]: This looks like a typo.

**Comment by gguuss** [comment]
@fhinkel I was going to finish migrating off of the Apiary client before moving to the separate repo.

File: iot/manager/manager.js (hunk 1 above)
Comment (gguuss) [reply iot/manager/manager.js]: Done.

File: iot/manager/manager.js (hunk 2 above)
Comment (gguuss) [reply iot/manager/manager.js]: Done

**Comment by gguuss** [comment]
Merging, after I have completed the migration we'll move this to the Cloud client library repo to simplify testing and releasing.

**Comment by gguuss** [comment]
I'm two PRs away from being ready to move to the Cloud client library repo.

**Pull request merged**